### PR TITLE
Get image info by remote path if file is external

### PR DESCRIPTION
**Pull request by golenkovm**
This PR covers Issue #180 .

### CODE REVIEW

File: classes/local/store/object_file_system.php [hunk 1]
```diff
@@ -753,4 +753,28 @@ public function add_file_from_string($content) {
...
+     */
+    public function get_imageinfo(stored_file $file) {
+        if (!$this->is_image_from_storedfile($file)) {
+            return false;
+        }
+
+        $contenthash = $file->get_contenthash();
+        $location = $this->get_object_location_from_hash($contenthash);
+
+        if ($location === OBJECT_LOCATION_EXTERNAL) {
```
Comment (dmitriim) [inline classes/local/store/object_file_system.php]: I'm a bit worried about '===' here. If location will be returned as string '2' this if will return false.

File: classes/local/store/object_file_system.php [hunk 2]
```diff
@@ -753,4 +753,28 @@ public function add_file_from_string($content) {
...
+
+    /**
+     * Returns information about image.
+     * Information is determined from the file content.
+     * If file is external get image info by remote path.
+     *
+     * @param stored_file $file The file to inspect
+     * @return mixed array with width, height and mimetype; false if not an image
+     */
+    public function get_imageinfo(stored_file $file) {
```
Comment (dmitriim) [inline classes/local/store/object_file_system.php]: Would be good to see some tests to cover this function

File: classes/local/store/object_file_system.php (hunk 1 above)
Comment (dmitriim) [reply classes/local/store/object_file_system.php]: Actually scratch that. I looked at the code and get_object_location_from_hash will never return a string

**Comment by tuanngocnguyen** [comment]
Hi @golenkovm ,

Since this has been inactive for some time, I am closing this PR.
Please create new MR against MOODLE_33_STABLE/MOODLE_310_STABLE if you are still working on the issue.
Thanks

**Pull request closed without merging**